notebooks/05_Kubeflow_Pipeline/05_04_Pipeline_SageMaker.ipynb

```diff
@@ -1,454 +1,612 @@
-   ]
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Mnist classification pipeline using Sagemaker\n",
+        "\n",
+        "This sample runs a pipeline to train a classficiation model using Kmeans with MNIST dataset on Sagemaker."
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "We will have all required steps here and for other details like how to get source data, please check [documentation](https://github.com/kubeflow/pipelines/tree/master/samples/contrib/aws-samples/mnist-kmeans-sagemaker).\n",
+        "\n",
+        "\n",
+        "This sample is based on the [Train a Model with a Built-in Algorithm and Deploy it](https://docs.aws.amazon.com/sagemaker/latest/dg/ex1.html).\n",
+        "\n",
+        "The sample trains and deploy a model based on the [MNIST dataset](http://www.deeplearning.net/tutorial/gettingstarted.html).\n",
+        "\n"
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Prerequisite"
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "1. Create an S3 bucket to store pipeline data\n",
+        "\n",
+        "> Note: Be sure to change the HASH variable to random hash and change AWS_REGION before running next cell\n",
+        "\n",
+        "> Note: you use us-east-1, please use command `!aws s3 mb s3://$S3_BUCKET --region $AWS_REGION --endpoint-url https://s3.us-east-1.amazonaws.com`"
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "!pip install sagemaker https://storage.googleapis.com/ml-pipeline/release/0.1.29/kfp.tar.gz --upgrade\n",
+        "!pip install boto3 --user\n",
+        "!pip show kfp"
+      ],
+      "outputs": [],
+      "execution_count": null,
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import random, string\n",
+        "HASH = ''.join([random.choice(string.ascii_lowercase) for n in range(16)] + [random.choice(string.digits) for n in range(16)])\n",
+        "AWS_REGION = 'us-west-2'\n",
+        "S3_BUCKET = '{}-kubeflow-pipeline-data'.format(HASH)\n",
+        "!aws s3 mb s3://$S3_BUCKET --region $AWS_REGION"
+      ],
+      "outputs": [],
+      "execution_count": null,
+      "metadata": {}
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "2. Copy dataset\n",
+        "\n",
+        "> Copy `data` and `valid_data.csv` into your S3 bucket."
+      ],
+      "metadata": {}
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import pickle, gzip, numpy, urllib.request, json\n",
+        "from urllib.parse import urlparse\n",
+        "\n",
+        "# Load the dataset\n",
+        "urllib.request.urlretrieve(\"http://deeplearning.net/data/mnist/mnist.pkl.gz\", \"mnist.pkl.gz\")\n",
```

Comment:
Can you check inference part. Let's try to reuse the same dataset for inference. Currently, inference prepare data one more time.

Reply:
Ok removed the code that downloaded the data since it was redundant.